code/common/checkinputs.q

```diff
@@ -0,0 +1,276 @@
+\d .checkinputs
+
+// checkinputs is the main function called when running a query - it checks:
+//   (i) input format
+//  (ii) whether any parameter pairs clash
+// (iii) parameter specific checks
+// The input dictionary accumulates some additional table information/inferred info
+checkinputs:{[dict]
+    dict:isdictionary dict;
+    if[in[`sqlquery;key dict];:isstring[dict;`sqlquery]];
+    dict:checkdictionary dict;
+    dict:checkinvalidcombinations dict;
+    dict:checkrepeatparams dict;
+    dict:checkeachparam[dict;1b];
+    dict:checkeachparam[dict;0b];
+    dict[`checksperformed]:1b;
+    :dict;
+  };
+
+checkdictionary:{[dict]
+    if[not checkkeytype dict;'`$"Input dictionary must have keys of type 11h"];
+    if[not checkrequiredparams dict;'`$.checkinputs.formatstring["Required parameters missing:{}";.checkinputs.getrequiredparams[]except key dict]];
+    if[not checkparamnames dict;'`$.checkinputs.formatstring["Invalid parameter present:{}. For a list of valid parameters, use the command .checkinputs.getvalidparams[]";key[dict]except .checkinputs.getvalidparams[]]];
+    :dict;
+  };
+
+isdictionary:{[dict]$[99h~type dict;:dict;'`$"Input must be a dictionary"]};
+checkkeytype:{[dict]11h~type key dict};
+checkrequiredparams:{[dict]all .checkinputs.getrequiredparams[]in key dict};
+getrequiredparams:{[]exec parameter from .checkinputs.checkinputsconfig where required}
+checkparamnames:{[dict]all key[dict]in .checkinputs.getvalidparams[]};
+getvalidparams:{[]exec parameter from .checkinputs.checkinputsconfig};
```

Comment:
Are these functions used frequently/outside of the above `checkinputs` & `checkdictionary` functions?

They're all quite short, this is probably debatable but IMO they're short enough that it's not worth turning these into separate functions unless they're being used a lot.